GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -125,91 +237,191 @@ static GVRAssimpImporter readFileFromResources(GVRContext gvrContext,
      *            Name of the file to import.
      * @return An instance of {@link GVRAssimpImporter}.
      */
-    static GVRAssimpImporter readFileFromSDCard(GVRContext gvrContext,
+    GVRAssimpImporter readFileFromSDCard(GVRContext gvrContext,
             String filename, EnumSet<GVRImportSettings> settings) {
         long nativeValue = NativeImporter.readFileFromSDCard(filename, GVRImportSettings.getAssimpImportFlags(settings));
         return new GVRAssimpImporter(gvrContext, nativeValue);
     }
 
-    static GVRModelSceneObject loadJassimpModel(final GVRContext context,
-            String filePath, GVRResourceVolume.VolumeType volumeType,
-            EnumSet<GVRImportSettings> settings) throws IOException {
-        return loadJassimpModel(context, filePath, volumeType, settings, false);
+    /**
+     * Loads a scene object {@link GVRModelSceneObject} from
+     * a 3D model and adds it to the scene.
+     *
+     * @param assetFile
+     *            A filename, relative to the root of the volume.
+     *
+     * @param volumeType
+     *            Where the asset is located (SD card, assets directory or network)
+     *            
+     * @param scene
+     *            If present, this asset loader will wait until all of the textures have been
+     *            loaded and then adds the model to the scene.
+     *            
+     * @return A {@link GVRModelSceneObject} that contains the meshes with textures and bones
+     * and animations.
+     * @throws IOException 
+     *
+     */
+    public GVRModelSceneObject loadModel(String filePath, GVRResourceVolume.VolumeType volumeType,
+            GVRScene scene) throws IOException
+    {
+        if (scene != null)
```

Comment:
The reason we have the scene is so that we can send the events somewhere (they go to the scene's event receiver) and so that we can add the model after all the textures are loaded. Which event receiver would be appropriate for the events if we don't have a scene?